### ISSUE TITLE
"Roles cannot inherit from classes," in language/objects may need more clarification

### ISSUE THREAD

**Issue by ronaldxs** [perl6/doc#1138]
If you have `class c {method m {}}; role r is c {}` then role `r` can execute method `m` both through instantiation and autopunning with `.new` and through auto creation of a class (but not instance?) with `r.m()`

The language/objects page claim "Roles cannot inherit from classes," under the inheritance subheading for roles seems confusing to me.  Watch out for a broken internal link in the index at the top of the page and try [https://docs.perl6.org/language/objects#Stubs](https://docs.perl6.org/language/objects#Stubs) to navigate just above the correct page section.

See IRC discussion: [https://irclog.perlgeek.de/perl6/2017-01-14#i_13920801](https://irclog.perlgeek.de/perl6/2017-01-14#i_13920801)

Bad link to "#Inheritance" seems to be issue #584

**Comment by jnthn** [comment]
Strictly speaking, the documentation is correct in so far as inheritance is something classes do, not roles.

When writing `role r is c { }` then the class(es) will be carried by the role, and - at the point when the role is composed into a class - the class will also gain these parent(s). The auto-punning of a role creates a class that does the role, and as part of composition the parent will be added to the class.

So, in summary, making a role inherit from a class means that any class the role is eventually composed in to will get that parent.

**Comment by ronaldxs** [comment]
I am new to roles and the doc section on punning makes it look like punning applies to instances and the section seems to me to make it a little hard to guess that punning applies to uninstantiated classes: [objects#Automatic_Role_Punning](https://docs.perl6.org/language/objects#Automatic_Role_Punning).  Perhaps a hint on class punning in that section might help.  While I am more than happy to take the word of those that know better that "Roles cannot inherit from classes", I am still uncertain of what observable behavior the phrase would mean for a Perl 6 user.

So I am suggesting at this point a mention of class punning.  I am also uncertain about the helpfulness of the phrase about roles not inheriting and wonder if it might not be better to remove it or refer to the updated section on punning.

**Comment by ronaldxs** [comment]
Just came across another case of roles being punned to an uninstantiated class:
```Perl6
role R {method m(){ say <hi>} }
class C is R {};
C.m;
```

**Comment by skids** [comment]
I think the above commit should address the rest of the remaining issues here?
